### PR TITLE
Problem: no trace detail on insufficient balance (backport #554)

### PR DESCRIPTION
**Pull request by mmsqe**
skip check

simple reproduce

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: tests/integration_tests/test_tracers.py [hunk 1]
```diff
@@ -156,6 +158,39 @@ def process(w3):
         assert res[0] == res[-1], res
 
 
+def test_trace_tx_reverse_transfer(ethermint):
```
Comment (yihuang) [inline tests/integration_tests/test_tracers.py:161]: can we do the comparison test with geth in ethermint integration test, if not easy, we can do that in cronos integration test instead.

File: tests/integration_tests/hardhat/contracts/FeeCollector.sol [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract FeeCollector {
+    event TokensMinted(address indexed to, uint256 amount);
+
+    function mint(uint256 amount) public payable {
+        emit TokensMinted(msg.sender, amount);
```
Comment (yihuang) [inline tests/integration_tests/hardhat/contracts/FeeCollector.sol:8]: should we also test the case that a "insufficient balance" transfer happens inside the contract?

File: tests/integration_tests/test_tracers.py (hunk 1 above)
Comment (mmsqe) [reply tests/integration_tests/test_tracers.py:161]: this test does flaky actually and gas is quite different for go ethereum

File: tests/integration_tests/hardhat/contracts/FeeCollector.sol (hunk 2 above)
Comment (mmsqe) [reply tests/integration_tests/hardhat/contracts/FeeCollector.sol:8]: transfer will trigger revert actually, but we can test STOP response for this `insufficient balance for transfer` vmError

Comment (yihuang) [reply tests/integration_tests/hardhat/contracts/FeeCollector.sol:8]: https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L186
can we trigger this code path inside contract?

Comment (mmsqe) [reply tests/integration_tests/hardhat/contracts/FeeCollector.sol:8]: yes
<Details>
{
    "height": "5",
    "txs_results": [
        {
            "code": 0,
            "data": "EvkDCicvZXRoZXJtaW50LmV2bS52MS5Nc2dFdGhlcmV1bVR4UmVzcG9uc2USzQMKQjB4MTY2ZGM1ZmIxM2M3MzFkYTMzZGViODkxYmRlZTdiZTA4OGFiNzQ5MTRiMzAyZmQ5NTYyY2IzOTM3NTE3YTcyNxLgAgoqMHg4Yzc2Y2ZjMTkzNGQ1MTIwY2M2NzNiNmU1ZGRmN2I4OGZlYjFjMThjEkIweDNmMmM5ZDU3YzA2ODY4NzgzNGYwZGU5NDJhOWJhYmI5ZTVhY2FiNTdkNTE2ZDM0ODBhM2MxNmVlMTY1YTQyNzMSQjB4MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwOGUxNTNmMjMzZDY0OTMwMDYzYjM1MTVmYjA3YWI2ZWNhNWYzNjZlNBogAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEIzcPOj6ZkgBSpCMHgwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwOkIweDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAo6LIBMiD+M+3Gk+lCkLxsul9O2OLqnNqtvSHtgIsNMv+7YZhdaA==",
            "log": "",
            "info": "",
            "gas_wanted": "22888",
            "gas_used": "22888",
            "events": [
                {
                    "type": "ethereum_tx",
                    "attributes": [
                        {
                            "key": "ethereumTxHash",
                            "value": "0x166dc5fb13c731da33deb891bdee7be088ab74914b302fd9562cb3937517a727",
                            "index": false
                        },
                        {
                            "key": "txIndex",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "coin_spent",
                    "attributes": [
                        {
                            "key": "spender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "1366091320137320aphoton",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "transfer",
                    "attributes": [
                        {
                            "key": "recipient",
                            "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "1366091320137320aphoton",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "tx",
                    "attributes": [
                        {
                            "key": "fee",
                            "value": "1366091320137320aphoton",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "action",
                            "value": "/ethermint.evm.v1.MsgEthereumTx",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "coin_spent",
                    "attributes": [
                        {
                            "key": "spender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "18633908679862681aphoton",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "coin_received",
                    "attributes": [
                        {
                            "key": "receiver",
                            "value": "ethm133mvlsvnf4gjpnr88dh9mhmm3rltrsvvdej90r",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "18633908679862681aphoton",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "transfer",
                    "attributes": [
                        {
                            "key": "recipient",
                            "value": "ethm133mvlsvnf4gjpnr88dh9mhmm3rltrsvvdej90r",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "18633908679862681aphoton",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "ethereum_tx",
                    "attributes": [
                        {
                            "key": "amount",
                            "value": "18633908679862681",
                            "index": false
                        },
                        {
                            "key": "ethereumTxHash",
                            "value": "0x166dc5fb13c731da33deb891bdee7be088ab74914b302fd9562cb3937517a727",
                            "index": true
                        },
                        {
                            "key": "txGasUsed",
                            "value": "22888",
                            "index": false
                        },
                        {
                            "key": "txHash",
                            "value": "48623CA7F1EA8155DF5BBDA14AE2F8AF7EB6497AD75A3FC514B5EB61E1AD370A",
                            "index": false
                        },
                        {
                            "key": "recipient",
                            "value": "0x8c76cfc1934d5120cc673b6e5ddf7b88feb1c18c",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "module",
                            "value": "evm",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "0x8e153f233d64930063b3515fb07ab6eca5f366e4",
                            "index": false
                        },
                        {
                            "key": "txType",
                            "value": "2",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                }
            ],
            "codespace": ""
        },
        {
            "code": 0,
            "data": "ErkBCicvZXRoZXJtaW50LmV2bS52MS5Nc2dFdGhlcmV1bVR4UmVzcG9uc2USjQEKQjB4ZGFjMDhmOGFkN2NkNjljNjRiZjIxNWE1NjI5N2JjZWI1MTY4NGRjNDJiNDQwM2NkNzdkNTNmMWU2Y2ExODdjMyIhaW5zdWZmaWNpZW50IGJhbGFuY2UgZm9yIHRyYW5zZmVyKJymATIg/jPtxpPpQpC8bLpfTtji6pzarb0h7YCLDTL/u2GYXWg=",
            "log": "",
            "info": "",
            "gas_wanted": "22888",
            "gas_used": "21276",
            "events": [
                {
                    "type": "ethereum_tx",
                    "attributes": [
                        {
                            "key": "ethereumTxHash",
                            "value": "0xdac08f8ad7cd69c64bf215a56297bceb51684dc42b4403cd77d53f1e6ca187c3",
                            "index": false
                        },
                        {
                            "key": "txIndex",
                            "value": "1",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "coin_spent",
                    "attributes": [
                        {
                            "key": "spender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "1366091320137320aphoton",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "transfer",
                    "attributes": [
                        {
                            "key": "recipient",
                            "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "1366091320137320aphoton",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "tx",
                    "attributes": [
                        {
                            "key": "fee",
                            "value": "1366091320137320aphoton",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "action",
                            "value": "/ethermint.evm.v1.MsgEthereumTx",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "coin_received",
                    "attributes": [
                        {
                            "key": "receiver",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "96213701855180aphoton",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "transfer",
                    "attributes": [
                        {
                            "key": "recipient",
                            "value": "ethm13c2n7geavjfsqcan290mq74kajjlxehyzhly4p",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                            "index": false
                        },
                        {
                            "key": "amount",
                            "value": "96213701855180aphoton",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "sender",
                            "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "ethereum_tx",
                    "attributes": [
                        {
                            "key": "amount",
                            "value": "18633908679862681",
                            "index": false
                        },
                        {
                            "key": "ethereumTxHash",
                            "value": "0xdac08f8ad7cd69c64bf215a56297bceb51684dc42b4403cd77d53f1e6ca187c3",
                            "index": true
                        },
                        {
                            "key": "txGasUsed",
                            "value": "21276",
                            "index": false
                        },
                        {
                            "key": "txHash",
                            "value": "F27ECC7606CB76881F90FA714A8A7249807748CDF53F9B6BD6E7BAFB1CB86A6A",
                            "index": false
                        },
                        {
                            "key": "recipient",
                            "value": "0x8c76cfc1934d5120cc673b6e5ddf7b88feb1c18c",
                            "index": false
                        },
                        {
                            "key": "ethereumTxFailed",
                            "value": "insufficient balance for transfer",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                },
                {
                    "type": "message",
                    "attributes": [
                        {
                            "key": "module",
                            "value": "evm",
                            "index": false
                        },
                        {
                            "key": "sender",
                            "value": "0x8e153f233d64930063b3515fb07ab6eca5f366e4",
                            "index": false
                        },
                        {
                            "key": "txType",
                            "value": "2",
                            "index": false
                        },
                        {
                            "key": "msg_index",
                            "value": "0",
                            "index": false
                        }
                    ]
                }
            ],
            "codespace": ""
        }
    ],
    "finalize_block_events": [
        {
            "type": "fee_market",
            "attributes": [
                {
                    "key": "base_fee",
                    "value": "51311279335",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coin_received",
            "attributes": [
                {
                    "key": "receiver",
                    "value": "ethm1m3h30wlvsf8llruxtpukdvsy0km2kum8nlf6yz",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coinbase",
            "attributes": [
                {
                    "key": "minter",
                    "value": "ethm1m3h30wlvsf8llruxtpukdvsy0km2kum8nlf6yz",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coin_spent",
            "attributes": [
                {
                    "key": "spender",
                    "value": "ethm1m3h30wlvsf8llruxtpukdvsy0km2kum8nlf6yz",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coin_received",
            "attributes": [
                {
                    "key": "receiver",
                    "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "transfer",
            "attributes": [
                {
                    "key": "recipient",
                    "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                    "index": false
                },
                {
                    "key": "sender",
                    "value": "ethm1m3h30wlvsf8llruxtpukdvsy0km2kum8nlf6yz",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "message",
            "attributes": [
                {
                    "key": "sender",
                    "value": "ethm1m3h30wlvsf8llruxtpukdvsy0km2kum8nlf6yz",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "mint",
            "attributes": [
                {
                    "key": "bonded_ratio",
                    "value": "0.999999917610992000",
                    "index": false
                },
                {
                    "key": "inflation",
                    "value": "0.129999949275418003",
                    "index": false
                },
                {
                    "key": "annual_provisions",
                    "value": "259999919971971492.453327518821438725",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coin_spent",
            "attributes": [
                {
                    "key": "spender",
                    "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coin_received",
            "attributes": [
                {
                    "key": "receiver",
                    "value": "ethm1jv65s3grqf6v6jl3dp4t6c9t9rk99cd8u3272a",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "transfer",
            "attributes": [
                {
                    "key": "recipient",
                    "value": "ethm1jv65s3grqf6v6jl3dp4t6c9t9rk99cd8u3272a",
                    "index": false
                },
                {
                    "key": "sender",
                    "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "41194501478stake",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "message",
            "attributes": [
                {
                    "key": "sender",
                    "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "commission",
            "attributes": [
                {
                    "key": "amount",
                    "value": "2018530572.422000000000000000stake",
                    "index": false
                },
                {
                    "key": "validator",
                    "value": "ethmvaloper18z6q38mhvtsvyr5mak8fj8s8g4gw7kjjgskm6l",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "rewards",
            "attributes": [
                {
                    "key": "amount",
                    "value": "20185305724.220000000000000000stake",
                    "index": false
                },
                {
                    "key": "validator",
                    "value": "ethmvaloper18z6q38mhvtsvyr5mak8fj8s8g4gw7kjjgskm6l",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "commission",
            "attributes": [
                {
                    "key": "amount",
                    "value": "2018530572.422000000000000000stake",
                    "index": false
                },
                {
                    "key": "validator",
                    "value": "ethmvaloper12luku6uxehhak02py4rcz65zu0swh7wjnrshgq",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "rewards",
            "attributes": [
                {
                    "key": "amount",
                    "value": "20185305724.220000000000000000stake",
                    "index": false
                },
                {
                    "key": "validator",
                    "value": "ethmvaloper12luku6uxehhak02py4rcz65zu0swh7wjnrshgq",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "BeginBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "coin_received",
            "attributes": [
                {
                    "key": "receiver",
                    "value": "ethm17xpfvakm2amg962yls6f84z3kell8c5lthdzgl",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "2635968938419460aphoton",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "EndBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "block_bloom",
            "attributes": [
                {
                    "key": "bloom",
                    "value": "@\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "EndBlock",
                    "index": false
                }
            ]
        },
        {
            "type": "block_gas",
            "attributes": [
                {
                    "key": "height",
                    "value": "5",
                    "index": false
                },
                {
                    "key": "amount",
                    "value": "44164",
                    "index": false
                },
                {
                    "key": "mode",
                    "value": "EndBlock",
                    "index": false
                }
            ]
        }
    ],
    "validator_updates": null,
    "consensus_param_updates": {
        "block": {
            "max_bytes": "1048576",
            "max_gas": "81500000"
        },
        "evidence": {
            "max_age_num_blocks": "100000",
            "max_age_duration": "172800000000000",
            "max_bytes": "1048576"
        },
        "validator": {
            "pub_key_types": [
                "ed25519"
            ]
        },
        "version": {},
        "abci": {}
    },
    "app_hash": "jsJdva6c7QjteORO1NSuqQTcIq/4OQMkof4FX0JhIFM="
}
</Details>

**Review by yihuang — APPROVED** [review]

**Pull request merged**